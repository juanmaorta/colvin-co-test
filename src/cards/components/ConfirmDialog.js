import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogTitle from '@material-ui/core/DialogTitle'

class ConfirmDialog extends Component {
  render () {
    const {
      cardId,
      handleAccept,
      handleClose,
      open,
      title
    } = this.props

    return (
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{title}</DialogTitle>
        <DialogActions>
          <Button onClick={() => handleAccept(cardId)} color='primary'>
              Accept
          </Button>
          <Button onClick={handleClose} color='primary' autoFocus>
              Cancel
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

ConfirmDialog.propTypes = {
  cardId: PropTypes.string,
  handleAccept: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  open: PropTypes.bool,
  title: PropTypes.string.isRequired
}

ConfirmDialog.defaultProps = {
  open: false
}

export default ConfirmDialog
