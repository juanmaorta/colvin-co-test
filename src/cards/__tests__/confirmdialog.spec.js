/* global expect, describe, it */
import React from 'react'
import { shallow } from 'enzyme'

import ConfirmDialog from '../components/ConfirmDialog'
import * as stubs from './stubs'

describe('ConfirmDialog snapshots', () => {
  it('should match for ConfirmDialog component', () => {
    const wrapper = shallow(
      <ConfirmDialog
        cardId={stubs.cardId}
        handleAccept={jest.fn}
        handleClose={jest.fn}
        open
        title='Some random title'
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
