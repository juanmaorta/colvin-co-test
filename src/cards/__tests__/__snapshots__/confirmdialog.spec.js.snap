// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog snapshots should match for ConfirmDialog component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmDialog
    cardId="3c6f4977-c58b-460f-98ad-acd2fae3a2ce"
    handleAccept={[Function]}
    handleClose={[Function]}
    open={true}
    title="Some random title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-describedby": "alert-dialog-description",
      "aria-labelledby": "alert-dialog-title",
      "children": Array [
        <WithStyles(DialogTitle)
          id="alert-dialog-title"
>
          Some random title
</WithStyles(DialogTitle)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Accept
          </WithStyles(Button)>
          <WithStyles(Button)
                    autoFocus={true}
                    color="primary"
                    onClick={[Function]}
          >
                    Cancel
          </WithStyles(Button)>
</WithStyles(DialogActions)>,
      ],
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Some random title",
          "id": "alert-dialog-title",
        },
        "ref": null,
        "rendered": "Some random title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Accept
</WithStyles(Button)>,
            <WithStyles(Button)
              autoFocus={true}
              color="primary"
              onClick={[Function]}
>
              Cancel
</WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Accept",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Accept",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "children": "Cancel",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "alert-dialog-description",
        "aria-labelledby": "alert-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="alert-dialog-title"
>
            Some random title
</WithStyles(DialogTitle)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Accept
            </WithStyles(Button)>
            <WithStyles(Button)
                        autoFocus={true}
                        color="primary"
                        onClick={[Function]}
            >
                        Cancel
            </WithStyles(Button)>
</WithStyles(DialogActions)>,
        ],
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Some random title",
            "id": "alert-dialog-title",
          },
          "ref": null,
          "rendered": "Some random title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Accept
</WithStyles(Button)>,
              <WithStyles(Button)
                autoFocus={true}
                color="primary"
                onClick={[Function]}
>
                Cancel
</WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Accept",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Accept",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "children": "Cancel",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
